src/main/scala/lms/core/stub.scala

```diff
@@ -904,6 +904,34 @@ object PrimitiveTypeLess {
       assert(t == y.t, s"t ${t} is not the same as y.t ${y.t}")
       BOOL(Adapter.g.reflect("<", x, y.x))
     }
+
+    def >(y: NUM)(implicit pos: SourceContext): BOOL = {
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t}")
+      BOOL(Adapter.g.reflect(">", x, y.x))
+    }
+
+    def tanh()(implicit pos: SourceContext): NUM = {
+      NUM(Adapter.g.reflect("tanh", x), t)
+    }
+
+    def tanh_grad(y: NUM, m: Manifest[_])(implicit pos: SourceContext): NUM = {
+      this * (NUM_ONE(m) - y * y)
+    }
```

Comment:
I think we probably don't need the `m: Manifest[_]` here, since you can get the manifest via `y.t`

Reply:
done.